docs/components/Header/Header.tsx

```diff
@@ -1,26 +1,59 @@
+import { useState, useEffect } from 'react';
 import Link from 'next/link';
 import styles from './Header.module.scss';
+import clsx from 'clsx';
 
-export const Header = () => (
-  <header className={styles.header}>
-    <nav>
-      <ul className={styles.navItems}>
-        <li>
-          <Link href="/">CooperTS</Link>
-        </li>
-        <li>
-          <Link href="/about">About</Link>
-        </li>
-        <li>
-          <Link href="/start">Getting Started</Link>
-        </li>
-        <li>
-          <Link href="/guide">Guide</Link>
-        </li>
-        <li>
-          <Link href="/examples">Examples</Link>
-        </li>
-      </ul>
-    </nav>
-  </header>
-);
+export const Header = () => {
+  let [isScrolled, setIsScrolled] = useState(false);
+
+  useEffect(() => {
+    function onScroll() {
+      setIsScrolled(window.scrollY > 0);
+    }
+    onScroll();
+    window.addEventListener('scroll', onScroll);
+    return () => {
+      window.removeEventListener('scroll', onScroll);
+    };
+  }, []);
+
+  return (
+    <header
+      className={clsx(
+        'sticky top-0 z-50 flex flex-wrap items-center justify-between bg-white px-4 py-5 shadow-md shadow-slate-900/5 transition duration-500 dark:shadow-none sm:px-6 lg:px-8',
+        {
```

Comment:
Tailwindcss makes the html into a design tool. During prototyping classNames are extremely verbose, but they are extremely flexible in this form. After refinement, these classNames are collapsed into a css "component" - a single className that extends each of these. 

The imported styles return in next phase (though tailwindcss classNames that are not huge lists like this tend to remain inline)